lib/Doctrine/ORM/Cache/DefaultEntityHydrator.php

```diff
@@ -141,7 +142,7 @@ public function loadCacheEntry(ClassMetadata $metadata, EntityCacheKey $key, Ent
 
             $assocClass  = $data[$name]->class;
             $assocId     = $data[$name]->identifier;
-            $isEagerLoad = ($assoc['fetch'] === ClassMetadata::FETCH_EAGER || ($assoc['type'] === ClassMetadata::ONE_TO_ONE && ! $assoc['isOwningSide']));
+            $isEagerLoad = ($assoc['fetch'] === ClassMetadata::FETCH_EAGER || ($assoc instanceof OneToOneAssociationMapping && ! $assoc->isOwningSide()));
```

Comment:
Why don't you use `isToOne()` here?

Reply:
:thinking: it wouldn't be equivalent, would it? `isToOne()` means either one to one or many to one.

Reply:
Oh right, sorry.